holidays/ical.py

```diff
@@ -0,0 +1,229 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+import uuid
+from datetime import date, datetime, timedelta, timezone
+from typing import Optional, Union
+
+from holidays.holiday_base import HolidayBase
+from holidays.version import __version__
+
+# iCal-specific constants
+CONTENT_LINE_MAX_LENGTH = 75
+CONTENT_LINE_DELIMITER = "\r\n"
+CONTENT_LINE_DELIMITER_WRAP = CONTENT_LINE_DELIMITER + " "
+
+
+class ICalExporter:
+    def __init__(self, holidays_object: HolidayBase, ical_timestamp: Optional[str] = None) -> None:
```

Comment:
Do we really need this DTSTAMP customization? 🙄

Reply:
Let's keep it simple then, now that we can probably offload advance `.ics` manipulation to `icalendar` package for advanced users

Reply:
Addressed in https://github.com/vacanza/holidays/pull/2362/commits/3e48a648c7102041ac4f1fac5152209475085d01 🙏